File: sdk/python/examples/smoke.py
"""Live read smoke test for the atradio Python SDK. Run: uv run examples/smoke.py"""

from atradio import AppView, favorite_rkey

av = AppView(None)  # None → default AppView base URL

recent = av.recent_stations(3)
print(f"recent_stations: {len(recent)} — first: {recent[0].station.name if recent else '-'}")

popular = av.popular_stations(3)
print(f"popular_stations: {len(popular)}")

play = av.global_recently_played(3)
print(f"global_recently_played: {len(play)}")

# Cross-SDK parity: the deterministic favorite key must match every other SDK.
rk = favorite_rkey("rb:00000000-0000-0000-0000-000000000000")
print(f"favorite_rkey(rb:0000…): {rk}  (len={len(rk)})")
assert len(rk) == 16, "favorite rkey must be a 16-char hex string"

print("Python read smoke OK")

File: sdk/ruby/examples/smoke.rb
# Live read smoke test for the atradio Ruby SDK. Run: ruby examples/smoke.rb
$LOAD_PATH.unshift File.expand_path("../lib", __dir__)
require "atradio"

recent = Atradio.recent_stations(3)
puts "recent_stations: #{recent.size} — first: #{recent.first&.dig('station', 'name')}"

popular = Atradio.popular_stations(3)
puts "popular_stations: #{popular.size}"

play = Atradio.global_recently_played(3)
puts "global_recently_played: #{play.size}"

# Cross-SDK parity: the deterministic favorite key must match every other SDK.
rk = Atradio.favorite_rkey("rb:00000000-0000-0000-0000-000000000000")
puts "favorite_rkey(rb:0000…): #{rk} (len=#{rk.length})"
raise "favorite rkey must be 16-char hex" unless rk.length == 16

puts "Ruby read smoke OK"
